src/drivers/ins/vectornav/libvnc/src/vn/sensors.c

```diff
@@ -2687,9 +2685,58 @@ VnError VnSensor_writeFirmwareUpdateRecord(VnSensor* s, char* record)
 	error = VnSensor_transactionNoFinalizeWithTiming(s, toSend, toSendLength, true, responseBuffer, &responseLength, 7000, 7000);
 	if (error != E_NONE)
 	{
-		char buffer[128];
-		memset(buffer, 0, sizeof(buffer));
-		strFromSensorError(buffer, error);
+		if (error >= E_SENSOR_HARD_FAULT && error <= E_SENSOR_ERROR_BUFFER_OVERFLOW)
```

Comment:
This is all code from the official VectorNav driver, I'd prefer we don't change it so that we can drop any new version on top. 

If you're interested we could get in touch with VectorNav and get the changes incorporated?

Reply:
This change is fine for now, but might be lost if/when we update the library from VectorNav.